Marlin/Marlin_main.cpp

```diff
@@ -4295,7 +4295,7 @@ inline void gcode_M109() {
     #define TEMP_CONDITIONS (!residency_start_ms || PENDING(now, residency_start_ms + (TEMP_RESIDENCY_TIME) * 1000UL))
   #else
     // Loop until the temperature is very close target
-    #define TEMP_CONDITIONS (isHeatingHotend(target_extruder))
+    #define TEMP_CONDITIONS (no_wait_for_cooling ? isHeatingHotend(target_extruder) : isCoolingHotend(target_extruder))
```

Comment:
@CONSULitAS you use the '_R_' parameter for cooling and waiting, not '_S_', also this feature has been in Marlin for years but is currently broken in RC/RCBugFix. If the user selects a temperature below room temp using the '_R_' parameter then that is user error imo. Regardless, it used to work and now it does not.